### DISCUSSION TITLE
Mongock support for spring reactive repositories

### DISCUSSION THREAD

**Discussion by sfgvieira** [flamingock/mongock#46] [Q&A] [answered]
Hello,

I would like to know if you still have plans to support spring reactive mongo repositories (there was a mention in the v4 docs but I don't see anything in v5 docs).

Thanks and happy new year! :)

**Comment by dieppa** [comment, marked as answer]
Hello @sfgvieira ,

**Short answer:** Yes 

**Long answer**
As you probably know, Mongock requires a synchronous process in the vast majority of scenarios(I cannot really think any which doesn't). Each changeUnit must be executed after(and only if) the previous one has been successfully executed. Also, in most of the cases, it's used as part of the application's startup and it's not desired to start the application unless Mongock has run successfully. So from that's perspective, there is not really point in providing reactive support,  it's not priority, and what's more, it's sensitive and maybe counterproductive as it needs to be blocked anyway.

However, JHipster requires this for  their reactive module. As they don't want to be importing both, the reactive and non-reactive, libraries. So we are working with them to have this.


**Note**
- As mentioned, even using the reactive repositories the operations need to e blocked. We are not sure how we are going to do this yet. It will be pretty flexible at the beginning, delegating on the developer to lock the operations inside the changeUnit. If we see that they struggle with this and we get many issues, we'll extend it to do it ourselves by default.

- Timing? We are not sure, We have some priorities for the beginning of 2022, I would say no less than a month. Meanwhile I suggest to keep using the blocking repositories.

**Comment by dieppa** [comment]
@sfgvieira Mongock now supports MongoDB reactive driver. Take a look [here](https://docs.mongock.io/v5/driver/mongodb-reactive/index.html)
